examples/configfile/custom/v1alpha1/register.go

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/scheme"
+)
+
+var (
```

Comment:
can we move this to a single file and/or compress this?  It's... a lot of stuff that's probably not strictly necessary just for the configuration.  I *think* we could even just compose a scheme on the fly maybe.  Non-blocking, but I'd kinda like to see something like

```go
type CustomConfig struct {
  metav1.TypeMeta
  cfg.ControllerConfiguration `json:",inline"`
  SomeCustomField int `json:"someCustomField"`
}
func (SomeCustomConfig) DefaultGV() schema.GroupVersion {
  return {...}
}
```

and then

```go
var myCustomConfig CustomConfig
cfg.FromFileOrDie("/some/path", &myCustomConfig) // more on this on the invocation of FromFileOrDie
```

and `FromFile` just type-asserts on

```go
type SchemelessObject interface {
  DefaultGV() schema.GroupVersion
}
```

Reply:
Then you could even just put it in main.go for small projects

Reply:
Doing it like this is really only useful if we have different versions of the configuration API.

Reply:
Yeah, the goal would be that we have multiple versions over time. This first one is being implemented as `v1alpha1` doing it this way will allow us to verify we're not missing fields or improperly organized.

@DirectXMan12 would using a SchemeLessObject style preclude this? I'm not sure what the implication would be here. 

Regarding consolidating this example, yes that's totally possible.

The goals of ComponentConfig are in general to provide ways to build versioned configuration files for w/e the components it sets up.

Reply:
Based on offline conversations with @pwittrock and @DirectXMan12 we ended up going a different UX where the loader can be dynamically populated to include the proper information and still uses the scheme provided to `options` removing the duplication.